csharp/src/Google.Protobuf/UnknownFieldSet.cs

```diff
@@ -135,6 +136,75 @@ public override int GetHashCode()
             return ret;
         }
 
+        internal static bool TryGetField(UnknownFieldSet set, int field, out UnknownField value)
```

Comment:
It seems that it should be enough to just provide the TryGetField method here. The other methods (such as TryGetLastVarint) are IMHO just syntactic sugar that bloats the suface of this class and they don't really belong here. Extracting the last value should be done by CustomOptions class (and UnknownFieldSet and UnknownField should only provide the minimum necessary functionality to allow that)